core/src/test/java/org/ehcache/util/ResourcePoolsMergerTest.java

```diff
@@ -76,4 +73,24 @@ public void testUpdatingHeapSizeToLargerThanDisk() {
       assertThat(iae.getMessage(), is("Updating resource pools leads authoritative tier being smaller than caching tier"));
     }
   }
+  
+  @Test
+  public void testUpdateResourceUnit() {
+    ResourcePoolsBuilder existingPoolsBuilder = ResourcePoolsBuilder.newResourcePoolsBuilder();
+    existingPoolsBuilder = existingPoolsBuilder.heap(20L, EntryUnit.ENTRIES).disk(500, MemoryUnit.MB);
+    ResourcePools existing = existingPoolsBuilder.build();
+
+    ResourcePoolsBuilder toBeUpdatedPoolsBuilder = ResourcePoolsBuilder.newResourcePoolsBuilder();
+    toBeUpdatedPoolsBuilder = toBeUpdatedPoolsBuilder.disk(2, MemoryUnit.GB);
```

Comment:
This test fails today but should pass in the future once ARC is implemented.
So I would replace it with a test checking that we do not try to switch a count based resource to a memory based one.